ctests/test_triton_special.cpp

```diff
@@ -0,0 +1,82 @@
+#include <gtest/gtest.h>
+#include "c10/util/Logging.h"
+#include "flag_gems/operators.h"
+#include "torch/torch.h"
+
+class EmbeddingTest : public ::testing::TestWithParam<
+                          std::tuple<int64_t, int64_t, int64_t, int64_t, int64_t, bool, torch::ScalarType>> {
+};
+TEST_P(EmbeddingTest, CompareWithPyTorch) {
+  const torch::Device device(torch::kCUDA, 0);
+  auto [EmbeddingSize, Batch, M, N, padding_idx, scale_grad_by_freq, dtype] = GetParam();
+  auto options = torch::TensorOptions().dtype(dtype).device(device);
+  auto indices =
+      torch::randint(0,
+                     EmbeddingSize,
+                     {Batch, M},
+                     torch::TensorOptions().device(device).dtype(torch::kLong).requires_grad(false));
+  auto embedding = torch::randn({EmbeddingSize, N}, options.requires_grad(true));
+  int64_t actual_padding_idx = padding_idx == -1 ? -1 : padding_idx;
```

Comment:
What is the purpose of `actual_padding_idx`? I believe it serves the same function as `padding_idx`.